src/main/ipv-stub/ipv-authorize.ts

```diff
@@ -15,38 +18,119 @@ export const handler: Handler = async (
     case "POST":
       return post(event);
     default:
-      logger.error("Method not allowed");
-      return {
-        statusCode: 405,
-        body: JSON.stringify({
-          message: "Method not allowed",
-        }),
-      };
+      return errorResponse(405, "Method not allowed");
   }
 };
 
-function get(_event: APIGatewayProxyEvent): APIGatewayProxyResult {
+async function get(
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> {
   logger.info("Reached the GET endpoint");
+
+  const headers: APIGatewayProxyEventHeaders = event.headers;
+  const bearerToken = headers["Authorization"];
+  if (!bearerToken) {
```

Comment:
Stylistic preference - but there seems to be quite a few if and try statements in this method. Could pull some of this logic out into separate methods to make it more readable.

Reply:
I personally think it's more readable as it is than if I refactored to use separate methods, as the methods would be very simple

Reply:
Leaving this for now - reviewing after the initial stubs work is finished.